plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/internal/macrolayout/KlighdDiagramLayoutConnector.java

```diff
@@ -655,6 +692,20 @@ private void applyLayout(final LayoutMapping mapping, final boolean suppressEdge
                 public Boolean caseElkNode(final ElkNode layoutNode) {
                     final KNode node = (KNode) element;
                     
+                    // Apply the nodeId and layerId that were set on the LGraph on the ElkGraph
+                    if (layoutNode.hasProperty(LAYERING_LAYER_ID) 
+                            && layoutNode.hasProperty(CROSSING_MINIMIZATION_POSITION_ID)) {
+                        node.setProperty(CROSSING_MINIMIZATION_POSITION_ID,
+                                layoutNode.getProperty(CROSSING_MINIMIZATION_POSITION_ID));
+                        node.setProperty(LAYERING_LAYER_ID, layoutNode.getProperty(LAYERING_LAYER_ID));
+                    }
+                    
+                    if (layoutNode.hasProperty(CURRENT_POSITION)) {
+                        final int position = layoutNode.getProperty(CURRENT_POSITION);
+                        node.setProperty(CURRENT_POSITION, position);
```

Comment:
Why can't we use the ELK equivalent of these options here?

Reply:
This does in fact seem odd to define ELK options again here. This seems to come from previous bachelor's theses, maybe the reasoning here was to not add a direct dependency from this macrolayout of KLighD to specific layout algorithms such as elk.layered?